### PR TITLE
chore: shim `fvm_shared::bigint`

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:
- [x] Avoid using `fvm_shared::bigint` when possible.
- [x] Move `bigint_ser` and `biguint_ser` to `forest_shim`. It can use the modules from `fvm_shared_v3`.

## Reference issue to close (if applicable)

Closes 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

**Comment by lemmih** [comment]
@elmattic Feel like taking this PR?

**Comment by elmattic** [comment]
Some questions:
- Should `weight: BigInt` in `BlockHeader` use `BigInt` from `forest_shim`?
- I didn't touch to `BigInt` in `vm_actor/interface`

**Comment by LesnyRumcajs** [comment]
> Some questions:
> 
> * Should `weight: BigInt` in `BlockHeader` use `BigInt` from `forest_shim`?
I believe so.

> * I didn't touch to `BigInt` in `vm_actor/interface`
Why?

**Comment by elmattic** [comment]
> > Some questions:
> > 
> > * Should `weight: BigInt` in `BlockHeader` use `BigInt` from `forest_shim`?
> 
> I believe so.

It's a bit hard because `forest_shim::bigint::BigInt` lacks of `FromStr`. In doubt I will keep the code like this.

> 
> > * I didn't touch to `BigInt` in `vm_actor/interface`
> 
> Why?

No reason. Done.

File: blockchain/chain/src/store/base_fee.rs [hunk 1]
```diff
@@ -90,7 +90,7 @@ where
     // Compute next base fee based on the current gas limit and parent base fee.
     let parent_base_fee = ts.blocks()[0].parent_base_fee();
     Ok(compute_next_base_fee(
-        parent_base_fee,
+        &parent_base_fee.clone(),
```
Comment (creativcoder) [inline blockchain/chain/src/store/base_fee.rs:93]: I think we don't need `&` `.clone()`, just parent_base_fee

**Review by creativcoder — APPROVED** [review]
Looking good!

File: blockchain/chain/src/store/base_fee.rs (hunk 1 above)
Comment (elmattic) [reply blockchain/chain/src/store/base_fee.rs]: Good catch! Done.

**Review by tyshko5 — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]
@elmattic How about adding a rule that would disallow using FVM bigint directly? This can be done via clippy configuration file [here](https://github.com/ChainSafe/forest/blob/main/.clippy.toml).

**Comment by elmattic** [comment]
> @elmattic How about adding a rule that would disallow using FVM bigint directly? This can be done via clippy configuration file [here](https://github.com/ChainSafe/forest/blob/main/.clippy.toml).

Won't there be issues with existing legit `fvm_shared` imports?

**Comment by LesnyRumcajs** [comment]
> > @elmattic How about adding a rule that would disallow using FVM bigint directly? This can be done via clippy configuration file [here](https://github.com/ChainSafe/forest/blob/main/.clippy.toml).
> 
> Won't there be issues with existing legit `fvm_shared` imports?

You can add `#[allow(clippy::disallowed_types)]` to the shim unit. To my understanding, that is the only place where `fvm_shared` `BigInt` should be directly used.

**Pull request merged**